tutorials/chapter1_annexB.md

```diff
@@ -0,0 +1,322 @@
+```
+
+**Threads.**
+
+- One for the main program (139959112914816).
+- One executed for publishing 3x messages in a row (139959104505600).
+- One for the slow listener (139959112898304).
+
+**TestEvent constructors.**
+
+None of the first event loop steps modify the event data. Just after the publishing of `(0,hello)`, `(1,hello)`, and `(2,hello)`:
+
+- 3x `constructorInvocations`, one for every message. Note that the loop reuses the same address (0x7f4ac4b4fd70).
+- 3x `copyInvocations`, when the data was passed _by value_ to the listener (note the different address 0x7f4ab8000b70, etc.).
+As such, the publisher changing of its data does not affect to the listeners'.\
+Note this is a kpsr internal copy, don't be confused with the listener registration (reference `&event`) at `include/slow_listener.h`
+
+```C++
+_subscriber.registerListener(listenerName, [](const TestEvent &event) {
+```
+
+- 3x `destructorInvocations`, one for each `constructorInvocations` after the publishing.
+
+When the listeners finish,
+
+- 6x `destructorInvocations`: the number was incrased with the 3x `copyInvocations`.
+So, in this example:
+
+<p align="center">
+constructorInvocations + copyInvocations = destructorInvocations
+</p>
+
+## Example B2: Adding the smart loop
+
+This example shares most of the code with example B1:
+
+```diff
+- const int poolSize{0};
++ const int poolSize{8};
+```
+
+```C++
+    kpsrPublisher = eventLoop.getPublisher<TestEvent>("TestEvent",
+                                                      poolSize,
+                                                      initializerFunction,
+                                                      eventClonerFunction);
+```
+
+Any non-zero pool size when getting the publisher enables the Klepsydra smart pool: this is a memory allocation that improves the publish efficiency. See the [Smart Memory Pool](./chapter1.md#smart-memory-pool) on the main tutorial for additional details.
+
+### Sample output
+
+These are the main differences compared with example B1:
+
+```text
+$ bin/kpsr_tutorial_chapter1_annexB_2
+
+Invocations counters updating after getPublisher
+        emptyConstructorInvocations 8
+Invocations counters updating after publishing
+        constructorInvocations      3
+        destructorInvocations       3
+Invocations counters updating after slow listeners finished
+        (none)
+```
+
+**Threads.**
+
+Same 3 threads than in the previous example: main, publisher, an listener.
+
+**TestEvent constructors.**
+
+When the publisher are created:
+
+- 8x `emptyConstructorInvocations`: the smart memory pool are populated at the selected size.
+
+Just after the publishing:
+
+- same 3x `constructorInvocations` and 3x `destructorInvocations` but,
+- 0x `copyInvocations`, because listener data was aready allocated at the 8x pool.
+Go to example B5 and B6 to study a smaller pool.
+
+When the listeners finish,
+
+- `destructorInvocations` keeps the same, only the pool needs to be freed.
```

Comment:
the pool is freed when the destructor of the publisher is called. you can verify this by putting everything inside an unnnamed namespace
```
int main (... ) 
{
   { 
   /// original code here
   }
   printInvocations();
   return 0;
}
```